app/controllers/api/v1/users_controller.rb

```diff
@@ -11,6 +11,20 @@ def index
         render json: users, status: :ok
       end
 
+      def destroy_self
+        authorize current_user, :destroy_self?
+
+        result = Users::DestroySelf.call(user: current_user, password: confirmation_password)
+
+        if result.success?
+          render json: { message: "Account deleted successfully" }, status: :ok
+        else
+          render json: { error: "Unable to delete account. Error: #{result.error}" }, status: :unprocessable_entity
```

Comment:
Looks like this line is not being tested, I think when you remove the approach using rescue here, we will get it solved